src/diff.c

```diff
@@ -1192,7 +1196,8 @@ int git_diff__from_iterators(
 	git_repository *repo,
 	git_iterator *old_iter,
 	git_iterator *new_iter,
-	const git_diff_options *opts)
+	const git_diff_options *opts,
+	bool *index_updated)
```

Comment:
Similarly, could the `out` type parameter move to the front of the list, after the `git_diff`?

Alternately, it might be nice to move the `index_updated` value into the `git_diff` structure itself.